docs/guide/migration.md

```diff
@@ -1,5 +1,27 @@
 # Migration
 
+## v3.0.0
+
+Caching has been completely refactored in Nuxt API Party v3. If you are using caching, please read the [caching documentation](./caching) to understand the new caching system.
+
+::: warning
+Caching behavior is now controlled by the upstream endpoint. If your endpoint does not support caching, the responses will not be stored in the browser's cache. This is a breaking change from previous versions, where caching was done regardless.
+
+The simplest way to ensure caching is enabled on the backend is to set the `Cache-Control: max-age=3600` header on your API responses, which will cache the response for 1 hour.
```

Comment:
I love this change in v3. However, I fear like we may loose users how have no control over external APIs they connected to. 

I myself have an API which I use with Nuxt API Party. However, I can't control the server's caching behavior. A Nitro hook might solve this, but maybe keeping the `cache: true` for backwards compatibility while encouraging developers to use the cache control header instead, might be the right approach here?

Reply:
For client caching, I wouldn't mind having a deprecated `cache: true` as long as there can be a way to do it without nuxt-api-party being involved. I'm thinking of a `fetch` option that can be used to specify a function that will execute a cached fetch.

I'm thinking an example usage could be something like this.

import cachedFetch from "my-cached-fetch-library"
```ts
await useApiData("/foo", {
  fetch: cachedFetch
})
```

Of course, this library would also be used by nuxt-api-party when `cache: true` is passed as well, but it won't be tree-shakable when no legacy caching is needed.

Reply:
I've gone ahead and included this in the code. In the future, we can deprecate `cache: true` in favor for either a separate package or a nuxt import